### PR TITLE
Adapt to latest origin-operator-registry image

### PR DESCRIPTION
**Pull request by yuvalturg**
/registry was removed from the latest origin-operator-registry image,
and the workdir was changed as well, this breaks the hco's bundle image
build.

Signed-off-by: Yuval Turgeman <yturgema@redhat.com>

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: deploy/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,17 @@
 FROM quay.io/openshift/origin-operator-registry:latest
 
-COPY deploy/olm-catalog /registry
+WORKDIR /registry
+
+COPY deploy/olm-catalog .
+
+USER root
```
Comment (orenc1) [inline deploy/Dockerfile:7]: Could you please elaborate on why is that necessary?

Comment (yuvalturg) [reply deploy/Dockerfile:7]: Sure, tl;dr - https://github.com/operator-framework/operator-registry/pull/430 introduced a change that breaks our image.  Previously /registry was created in the operator-registry image, and had g+rwx permissions, so we could create and use /registry/bundles.db.  This is not the case anymore, so when running initializer, the bundles.db can't be written to /registry.  In addition to that, there is not /bin/sh, so RUN should be used with exec and not with `sh -c` - they are working on a fix in https://github.com/operator-framework/operator-registry/pull/435 but with this change, we won't depend on /bin/sh

File: deploy/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,17 @@
 FROM quay.io/openshift/origin-operator-registry:latest
 
-COPY deploy/olm-catalog /registry
+WORKDIR /registry
+
+COPY deploy/olm-catalog .
```
Comment (orenc1) [inline deploy/Dockerfile:5]: Isn't it the same result? Why did you split `COPY deploy/olm-catalog /registry` into two dockerfile commands?

Comment (yuvalturg) [reply deploy/Dockerfile:5]: Yes and no... yes because the end result will be the same - having /registry/kubevirt-hyperconverged, and no because we need WORKDIR to be set for the RUN and ENTRYPOINT (it was set previously on the operator-registry as well), but it was removed.  The alternative would be to keep the COPY command as it is, and move WORKDIR after, does it matter ?

Comment (orenc1) [reply deploy/Dockerfile:5]: /lgtm

**Review by tiraboschi — APPROVED** [review]
